src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Extensions/LoggerSinkConfigurationExtensions.cs

```diff
@@ -294,5 +406,127 @@ public static class LoggerSinkConfigurationExtensions
+        /// </summary>
+        /// <remarks>
+        ///     Supported telemetry types are Traces, Dependencies, Events, Requests and Metrics for which we provide extensions on <see cref="ILogger" />.
+        /// </remarks>
+        /// <param name="loggerSinkConfiguration">The logger configuration.</param>
+        /// <param name="serviceProvider">
+        ///     The provider instance to retrieve the <see cref="TelemetryClient"/> in the application services.
+        ///     Note that this is only required when the application requires W3C correlation.
+        /// </param>
+        /// <param name="connectionString">The required Application Insights connection string.</param>
+        /// <param name="restrictedToMinimumLevel">The minimum log event level required in order to write an event to the sink.</param>
+        /// <param name="configureOptions">The optional function to configure additional options to influence the behavior of how the telemetry is logged to Azure Application Insights.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="loggerSinkConfiguration"/> is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="connectionString"/> is blank.</exception>
+        public static LoggerConfiguration AzureApplicationInsightsWithConnectionString(
+            this LoggerSinkConfiguration loggerSinkConfiguration,
+            IServiceProvider serviceProvider,
+            string connectionString,
+            LogEventLevel restrictedToMinimumLevel,
+            Action<ApplicationInsightsSinkOptions> configureOptions)
+        {
+            Guard.NotNull(loggerSinkConfiguration, nameof(loggerSinkConfiguration), "Requires a logger configuration to add the Azure Application Insights sink to");
+            Guard.NotNullOrWhitespace(connectionString, nameof(connectionString), "Requires an instrumentation key to authenticate with Azure Application Insights while sinking telemetry");
+
+            var options = new ApplicationInsightsSinkOptions();
+            configureOptions?.Invoke(options);
+
+            var client = serviceProvider.GetService<TelemetryClient>();
+            if (client is null)
+            {
+                throw new InvalidOperationException(
+                    "Could not retrieve Microsoft telemetry client from the application registered services, this happens when the Application Insights services are not registered in the application services," 
+                    + "please use one of Arcus' extensions like 'services.AddHttpCorrelation()' to automatically register the Application Insights when using the W3C correlation system, "
+                    + $"when using the Hierarchical correlation system, use the {nameof(AzureApplicationInsightsWithConnectionString)} extension without the service provider instead");
+            }
```

Comment:
OK, this is an answer to one of my previous comments :)

Reply:
Just wondering if we should throw, and not create a new TelemetryClient behind the scenes.  The current implementation is the most 'pure'.  Creating a new TelemetryClient is maybe less 'aggressive' ?

Reply:
Don't get me started on 'pure' in C# 😄 .
If we would create it ourselves, then we implicitly use the same functionality as in our existing extensions overloads. This could may cause unexpected results as in that case built-in Microsoft dependencies may not get tracked as expected; or it would differ (like application version/name different).